Update GCC and Clang to latest available

milljm opened the pull request:
Attempt a GCC 8.2.0, and LLVM 7.0.1 build.
Remove CPPUNIT.
Fix miniconda/pkgs/urls.txt missing, causing permission errors when a user attempts to create an environment.
Fix chown issue after profile is created.
Remove make_payload.sh. That script no longer does anything as far as I can tell.

Closes #186 #188

Commit:
Update GCC and Clang to latest available

Attempt a GCC 8.2.0, and LLVM 7.0.1 build.
Remove CPPUNIT.
Fix miniconda/pkgs/urls.txt missing, causing permission errors when a user attempts to create an environment.
Fix chown issue after profile is created.
Remove make_payload.sh. That script no longer does anything as far as I can tell.

Closes #186 #188

milljm commented:
Looks like Darwin is going to run into trouble with the latest LLVM release. Possible lld usage.

Commit:
Appears lld does not yet work on Darwin platforms

Create symbolic link to lld only on linux machines for now.

milljm commented:
Once this: https://github.com/roystgnr/MetaPhysicL/pull/41 makes into moose's list of submodules, more things will start to pass.

Commit:
Support multiple shell environments

bash, sh, zsh, csh, tsch supported.

Do not modify user-owned environment (linux only). Instead, add
modules to the system-wide environment. Inform the user which
modules they should load. This will make new installations easier
when a user is unfamiliar with managing their own profiles.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/288646/) on 831f8ce wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/189/site)

This comment will be updated on new commits.

Commit:
Update how we handle the code sign cert

Check for the presence of moose-codesign before attempting
to delete it.

For now, see if we even need the code-signing cert. This is
TEST! I suspect, the get-task-allow entittlement is all that
is required when using LLDB (any LLDB. even ours).

Commit:
Force loading of miniconda module

When using moose-dev-gcc, on a darwin machine, one must also
load the miniconda module due to python-config returning clang-
only cflags.

Closes #191

Commit:
re-enable moose-codesign cert

I neglected to realize codesigning is not intended for the apps being built
by our LLVM compiler, but for our LLVM debugger to actually operate on the
users machine.

Commit:
Deal with lsb_release 'No LSB modules'

We shouldn't consider stderr being a non-null value as an indication
as to a general non-zero exit code. Use poll instead.

Debian `stretch` has no solution for 'No LSB modules are available'.

Commit:
Automake patch which should fix the help2man

We are experiencing this bug on Ubuntu-18:
http://gnu-automake.7480.n7.nabble.com/bug-30631-Automake-1-6-fails-to-build-with-perl-5-18-2-td22865.html

1.16.1 contains the patch fix.

milljm commented:
will be applied in rolling update #193
